File: apple/Sources/ReactNativeWatchOS/Modifiers/InteractiveDismissDisabled.swift
import SwiftUI

/// SwiftUI `.interactiveDismissDisabled(_:)`. Conditionally prevents
/// interactive (swipe) dismissal of the presentation it is applied to.
/// `value` defaults to true. Available on watchOS 9+.
enum RNWInteractiveDismissDisabledModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("interactiveDismissDisabled") { view, params, _ in
            AnyView(view.interactiveDismissDisabled(params.bool("value") ?? true))
        }
    }
}

File: src/modifiers/interactiveDismissDisabled.ts
import { createModifier } from './createModifier';

/// SwiftUI `.interactiveDismissDisabled(_:)`. Conditionally prevents an
/// interactive (swipe / drag) dismissal of the presentation it is applied
/// to — typically a `.sheet`. Pass `false` to re-enable interactive
/// dismissal. Available on watchOS 9+.
export function interactiveDismissDisabled(value = true) {
  return createModifier('interactiveDismissDisabled', { value });
}
